### PR TITLE
add recommended workflow for copier updates

### PR DESCRIPTION
**Pull request by gilesknap**
For projects that have not been updated for a long time I found the documented workflow helped avoid confusion of changing the template and tool versions simultaneously.

### CODE REVIEW

File: docs/how-to/update-template.md [hunk 1]
```diff
@@ -15,3 +17,24 @@ git diff --check
...
+## Recommended Workflow
+
+The following steps are recommended to update your project, especially for infreqently updated projects:
+
+- first make sure all the tools are updated to latest versions
+    - for devcontainers
+        - `ctrl+shift+p` -> `Remote-Containers: Rebuild Without Cache and Reopen in Container`
+    - for local development
+        - `pip install -e .[dev] --force-reinstall`
+- validate your project agains the latest tools
```
Comment (GDYendell) [inline docs/how-to/update-template.md:30]: against

Comment (gilesknap) [reply docs/how-to/update-template.md]: Thanks, Gary. Fixed.

File: docs/how-to/update-template.md [hunk 2]
```diff
@@ -15,3 +17,24 @@ git diff --check
 ```
 
 Once they are all fixed, make a commit.
+
+## Recommended Workflow
+
+The following steps are recommended to update your project, especially for infreqently updated projects:
```
Comment (GDYendell) [inline docs/how-to/update-template.md:23]: infrequently

**Comment by gilesknap** [comment]
> Did just updating dependencies and fixing before doing the update make it easier in some case? I always just update and fix everything in one go.

So my assertion is that if there is a long break between updates. Having both new ruff (or other) errors to contend with and the changes themselves was too much.

I guess this came about because I was updating to latest from an early beta version of copier template.

When the checks don't even run because of fundamental issues (in my case a missing _version.py) then it's difficult to know where the problem is coming from.

Perhaps I'm over thinking this though. I guess fixes for both causes of error can be performed together. My particular scenario was extreme.

I'm reasonably happy to close without merging.

**Review by coretl — APPROVED** [review]

**Review by GDYendell — APPROVED** [review]
Squash and merge?

**Comment by coretl** [comment]
I disabled it for the repo as we sometimes adopt from a branch and squashing loses the update ability. This doesn't apply here, so will re-enable it just for this PR...

**Pull request merged**